lib/daru/index.rb

```diff
@@ -84,6 +84,40 @@ def [](key, *rest)
       end
     end
 
+    def respond? *indexes
+      indexes.all? { |i| to_a.include?(i) || (i.is_a?(Numeric) && i < size) }
+    end
+
+    def pos *args
+      args = preprocess_range(args.first) if args.first.is_a? Range
+
+      if args.size == 1
+        self[args.first]
+      else
+        args.map { |index| by_single_key index }
+      end
+    end
+
+    def subset *args
+      if args.first.is_a? Range
+        slice args.first.begin, args.first.end
+      elsif include? args.first
+        # Assume args contain indexes not positions
+        Daru::Index.new args
+      else
+        # Assume args contain positions not indexes
+        Daru::Index.new args.map { |k| key k }
+      end
+    end
+
+    def at *args
+      if args.size == 1
+        key(args.first)
+      else
+        self.class.new(args.map { |pos| key pos })
+      end
```

Comment:
BTW², `args` is an appropriate name for arguments list only if there are many meanings arguments can have (which is a "code smell" on its own). In cases like this, the name like "positions" will suit much better.